internal/stores/contracts.go

```diff
@@ -294,6 +364,22 @@ func (s *SQLStore) RemoveContract(id types.FileContractID) error {
 	})
 }
 
+func (s *SQLStore) contract(id types.FileContractID) (dbContract, error) {
+	return contract(s.db, id)
+}
+
+func (s *SQLStore) contracts(set string) (contracts []dbContract, err error) {
+	err = s.db.
+		Model(&dbContract{}).
+		Preload("Host.Announcements").
+		Joins("INNER JOIN contract_set_contracts csc ON csc.db_contract_id = contracts.id").
+		Joins("INNER JOIN contract_sets cs ON cs.id = csc.db_contract_set_id").
+		Where("cs.name = ?", set).
+		Find(&contracts).
```

Comment:
Is this faster than something like `s.db.Where(dbContractSet{name: set}).Preload("Contracts.Host.Announcements").Take(&set)`?

Reply:
I couldn't make the `gorm` syntax work. I think I'm not properly selecting the right `Model`? The query you posted doesn't work either by the way. (I know it was pseudo code but still). I figured it was syntactic sugar for what I wrote anyway, that and the `INNER JOIN` is probably faster than the default join gorm uses so it's definitely not slower.